File: app/src/main/java/com/productivity/habits/ui/common/HabitIconRegistry.kt
package com.productivity.habits.ui.common

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.DirectionsRun
import androidx.compose.material.icons.automirrored.filled.DirectionsWalk
import androidx.compose.material.icons.automirrored.filled.MenuBook
import androidx.compose.material.icons.filled.AutoAwesome
import androidx.compose.material.icons.filled.Bedtime
import androidx.compose.material.icons.filled.Book
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Code
import androidx.compose.material.icons.filled.Edit
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.FitnessCenter
import androidx.compose.material.icons.filled.FlashOn
import androidx.compose.material.icons.filled.LocalCafe
import androidx.compose.material.icons.filled.MusicNote
import androidx.compose.material.icons.filled.Psychology
import androidx.compose.material.icons.filled.Schedule
import androidx.compose.material.icons.filled.SentimentSatisfied
import androidx.compose.material.icons.filled.Star
import androidx.compose.material.icons.filled.TrackChanges
import androidx.compose.material.icons.filled.WaterDrop
import androidx.compose.material.icons.filled.WbSunny
import androidx.compose.material.icons.filled.Work
import androidx.compose.ui.graphics.vector.ImageVector

data class HabitIconItem(
    val key: String,
    val label: String,
    val icon: ImageVector
)

object HabitIconRegistry {

    private val ICONS_MAP: Map<String, ImageVector> = mapOf(
        // Default category icons
        "activity" to Icons.AutoMirrored.Filled.DirectionsRun,
        "brain" to Icons.Default.Psychology,
        "book-open" to Icons.AutoMirrored.Filled.MenuBook,
        "book" to Icons.Default.Book,
        "zap" to Icons.Default.FlashOn,
        "heart" to Icons.Default.Favorite,
        "clock" to Icons.Default.Schedule,

        // Common habit icons
        "check" to Icons.Default.CheckCircle,
        "star" to Icons.Default.Star,
        "target" to Icons.Default.TrackChanges,
        "droplet" to Icons.Default.WaterDrop,
        "water" to Icons.Default.WaterDrop,
        "footprints" to Icons.AutoMirrored.Filled.DirectionsWalk,
        "walk" to Icons.AutoMirrored.Filled.DirectionsWalk,
        "run" to Icons.AutoMirrored.Filled.DirectionsRun,
        "dumbbell" to Icons.Default.FitnessCenter,
        "fitness" to Icons.Default.FitnessCenter,
        "moon" to Icons.Default.Bedtime,
        "sun" to Icons.Default.WbSunny,
        "coffee" to Icons.Default.LocalCafe,
        "code" to Icons.Default.Code,
        "sparkles" to Icons.Default.AutoAwesome,
        "smile" to Icons.Default.SentimentSatisfied,
        "music" to Icons.Default.MusicNote,
        "edit" to Icons.Default.Edit,
        "work" to Icons.Default.Work
    )

    val AVAILABLE_ICONS: List<HabitIconItem> = listOf(
        HabitIconItem("activity", "Activity", Icons.AutoMirrored.Filled.DirectionsRun),
        HabitIconItem("brain", "Mindfulness", Icons.Default.Psychology),
        HabitIconItem("book-open", "Reading", Icons.AutoMirrored.Filled.MenuBook),
        HabitIconItem("zap", "Productivity", Icons.Default.FlashOn),
        HabitIconItem("heart", "Health / Heart", Icons.Default.Favorite),
        HabitIconItem("clock", "Routine / Time", Icons.Default.Schedule),
        HabitIconItem("droplet", "Hydration", Icons.Default.WaterDrop),
        HabitIconItem("dumbbell", "Fitness", Icons.Default.FitnessCenter),
        HabitIconItem("footprints", "Walking / Steps", Icons.AutoMirrored.Filled.DirectionsWalk),
        HabitIconItem("moon", "Sleep / Night", Icons.Default.Bedtime),
        HabitIconItem("sun", "Morning", Icons.Default.WbSunny),
        HabitIconItem("coffee", "Break / Focus", Icons.Default.LocalCafe),
        HabitIconItem("code", "Coding / Tech", Icons.Default.Code),
        HabitIconItem("target", "Target / Goals", Icons.Default.TrackChanges),
        HabitIconItem("sparkles", "Motivation", Icons.Default.AutoAwesome),
        HabitIconItem("star", "Star / Priority", Icons.Default.Star),
        HabitIconItem("music", "Music / Art", Icons.Default.MusicNote),
        HabitIconItem("smile", "Mood / Well-being", Icons.Default.SentimentSatisfied),
        HabitIconItem("edit", "Writing / Journal", Icons.Default.Edit),
        HabitIconItem("work", "Work / Projects", Icons.Default.Work)
    )

    fun getIcon(key: String?): ImageVector {
        if (key.isNullOrBlank()) return Icons.Default.CheckCircle
        val normalized = key.lowercase().trim()
        return ICONS_MAP[normalized] ?: Icons.Default.CheckCircle
    }
}

File: lib/ui/common/habit_icon_registry.dart
import 'package:flutter/material.dart';

class HabitIconItem {
  final String key;
  final String label;
  final IconData icon;

  const HabitIconItem({
    required this.key,
    required this.label,
    required this.icon,
  });
}

class HabitIconRegistry {
  static const Map<String, IconData> iconsMap = {
    // Default category icons
    'activity': Icons.directions_run,
    'brain': Icons.psychology,
    'book-open': Icons.menu_book,
    'book': Icons.book,
    'zap': Icons.bolt,
    'heart': Icons.favorite,
    'clock': Icons.schedule,

    // Badges and gamification icons
    'flame': Icons.local_fire_department,
    'shield': Icons.shield,
    'award': Icons.emoji_events,
    'trophy': Icons.emoji_events,
    'crown': Icons.auto_awesome,
    'medal': Icons.military_tech,
    'layers': Icons.layers,
    'compass': Icons.explore,
    'calendar': Icons.calendar_month,
    'trending_up': Icons.trending_up,
    'briefcase': Icons.work,
    'check_circle': Icons.check_circle,

    // Common habit icons
    'check': Icons.check_circle,
    'star': Icons.star,
    'target': Icons.track_changes,
    'droplet': Icons.water_drop,
    'water': Icons.water_drop,
    'footprints': Icons.directions_walk,
    'walk': Icons.directions_walk,
    'run': Icons.directions_run,
    'dumbbell': Icons.fitness_center,
    'fitness': Icons.fitness_center,
    'moon': Icons.bedtime,
    'sun': Icons.wb_sunny,
    'coffee': Icons.local_cafe,
    'code': Icons.code,
    'sparkles': Icons.auto_awesome,
    'smile': Icons.sentiment_satisfied,
    'music': Icons.music_note,
    'edit': Icons.edit,
    'work': Icons.work,
  };

  static const List<HabitIconItem> availableIcons = [
    HabitIconItem(key: 'activity', label: 'Activity', icon: Icons.directions_run),
    HabitIconItem(key: 'brain', label: 'Mindfulness', icon: Icons.psychology),
    HabitIconItem(key: 'book-open', label: 'Reading', icon: Icons.menu_book),
    HabitIconItem(key: 'zap', label: 'Productivity', icon: Icons.bolt),
    HabitIconItem(key: 'heart', label: 'Health / Heart', icon: Icons.favorite),
    HabitIconItem(key: 'clock', label: 'Routine / Time', icon: Icons.schedule),
    HabitIconItem(key: 'droplet', label: 'Hydration', icon: Icons.water_drop),
    HabitIconItem(key: 'dumbbell', label: 'Fitness', icon: Icons.fitness_center),
    HabitIconItem(key: 'footprints', label: 'Walking / Steps', icon: Icons.directions_walk),
    HabitIconItem(key: 'moon', label: 'Sleep / Night', icon: Icons.bedtime),
    HabitIconItem(key: 'sun', label: 'Morning', icon: Icons.wb_sunny),
    HabitIconItem(key: 'coffee', label: 'Break / Focus', icon: Icons.local_cafe),
    HabitIconItem(key: 'code', label: 'Coding / Tech', icon: Icons.code),
    HabitIconItem(key: 'target', label: 'Target / Goals', icon: Icons.track_changes),
    HabitIconItem(key: 'sparkles', label: 'Motivation', icon: Icons.auto_awesome),
    HabitIconItem(key: 'star', label: 'Star / Priority', icon: Icons.star),
    HabitIconItem(key: 'music', label: 'Music / Art', icon: Icons.music_note),
    HabitIconItem(key: 'smile', label: 'Mood / Well-being', icon: Icons.sentiment_satisfied),
    HabitIconItem(key: 'edit', label: 'Writing / Journal', icon: Icons.edit),
    HabitIconItem(key: 'work', label: 'Work / Projects', icon: Icons.work),
  ];

  static IconData getIcon(String? key) {
    if (key == null || key.trim().isEmpty) {
      return Icons.check_circle;
    }
    final normalized = key.toLowerCase().trim();
    return iconsMap[normalized] ?? Icons.check_circle;
  }
}
